### PR TITLE
Mark guild Role position as computed

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
Mark the position of a role as computed. I find it is easier to create roles with terraform and then move them into position manually.

Currently, I have to use
```hcl
    lifecycle {
        ignore_changes = [position]
    }
```
and this would alleviate that.

Shouldn't be a breaking change but also might be worth releasing as a 2.0

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Visit https://goreleaser.com for documentation on how to customize this
 # behavior.
+version: 2
```
Comment (Lucky3028) [inline .goreleaser.yml:3]: Please make this change in other PR.

**Review by Lucky3028 — CHANGES_REQUESTED** [review]
thx for your PR

File: discord/resource_discord_role.go [hunk 2]
```diff
@@ -62,10 +62,10 @@ func resourceDiscordRole() *schema.Resource {
 			"position": {
 				Type:         schema.TypeInt,
 				Optional:     true,
-				Default:      1,
+				Computed:     true,
 				ForceNew:     false,
 				ValidateFunc: validation.IntAtLeast(1),
-				Description:  "The position of the role. This is reverse indexed, with `@everyone` being `0`.",
+				Description:  "Position of the role. This is reverse indexed, with `@everyone` being `0`.",
```
Comment (Lucky3028) [inline discord/resource_discord_role.go:69]: nits: Please make changes in other commit.
(Same PR is OK)

File: docs/resources/role.md [hunk 3]
```diff
@@ -38,7 +38,7 @@ resource "discord_role" "moderator" {
 - `hoist` (Boolean) Whether the role should be hoisted. (default `false`)
 - `mentionable` (Boolean) Whether the role should be mentionable. (default `false`)
 - `permissions` (Number) The permission bits of the role.
-- `position` (Number) The position of the role. This is reverse indexed, with `@everyone` being `0`.
+- `position` (Number) Position of the role. This is reverse indexed, with `@everyone` being `0`.
```
Comment (Lucky3028) [inline docs/resources/role.md:41]: nits: Please make changes in other commit.
(Same PR is OK)

**Review by Lucky3028 — CHANGES_REQUESTED** [review]

**Review by Lucky3028 — COMMENTED** [review]
I think some people may want to set the roll position manually.
This PR is too destructive, isn’t it?

**Comment by Cyb3r-Jak3** [comment]
> I think some people may want to set the roll position manually.

You are still able to set the position of the role. By swapping to from Default to Computed, roles that have their position not set and are updated remotely won't trigger changes. If you have a value set for the position attribute, then changes will still be triggered.

https://github.com/hashicorp/terraform/issues/21278

**Comment by Lucky3028** [comment]
I see. Thanks!

**Review by Lucky3028 — APPROVED** [review]

**Pull request merged**